ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -181,17 +170,93 @@ S3CustomizeFileSystem(WT_STORAGE_SOURCE *storageSource, WT_SESSION *session, con
 static int
 S3FileSystemTerminate(WT_FILE_SYSTEM *fileSystem, WT_SESSION *session)
 {
-    S3_FILE_SYSTEM *fs;
+    S3_FILE_SYSTEM *fs = (S3_FILE_SYSTEM *)fileSystem;
 
     UNUSED(session); /* unused */
 
-    fs = (S3_FILE_SYSTEM *)fileSystem;
     delete (fs->connection);
     free(fs);
 
     return (0);
 }
 
+/*
+ * S3ObjectList --
+ *     Return a list of object names for the given location.
+ */
+static int
+S3ObjectList(WT_FILE_SYSTEM *fileSystem, WT_SESSION *session, const char *bucket,
+  const char *prefix, char ***objectList, uint32_t *count)
+{
+    S3_FILE_SYSTEM *fs = (S3_FILE_SYSTEM *)fileSystem;
+    std::vector<std::string> objects;
+    int ret;
+    if (ret = fs->connection->ListObjects(std::string(bucket), std::string(prefix), objects) != 0)
```

Comment:
We don't need to explicitly construct a `string` for `bucket` and `prefix` as there already is a `std::string` constructor that takes a `const char *`, see [the docs](https://www.cplusplus.com/reference/string/string/string/) for reference.